src/ring_theory/dedekind_domain/ideal.lean

```diff
@@ -586,11 +586,12 @@ noncomputable instance fractional_ideal.comm_group_with_zero :
   mul_inv_cancel := λ I, fractional_ideal.mul_inv_cancel,
   .. fractional_ideal.comm_semiring }
 
```

Comment:
Are we missing
```
instance : cancel_comm_monoid_with_zero (fractional_ideal A⁰ K) :=
{ ..fractional_ideal.comm_semiring,
  ..comm_group_with_zero.cancel_comm_monoid_with_zero }
```
That would make the old proof below slightly more computable; and I would guess the assumptions could be weakened if we prove it directly.

Reply:
Adding a computable `cancel_comm_monoid_with_zero` instance should work I think, I just don't know if there is any (mathematically interesting) case where fractional ideals are cancellative but don't have inverses, since `I⁻¹` is always defined, it just might not be an inverse.

To be clear, I don't want to go back to the previous `cancel_comm_monoid_with_zero` instance on ideals, since it still won't have good unification performance.